plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx

```diff
@@ -91,6 +101,28 @@ export const EntityOwnerPicker = () => {
     [backendEntities],
   );
 
+  const {
+    loading,
+    error,
+    value: ownerEntities,
+  } = useAsync(async () => {
+    const { items } = await catalogApi.getEntitiesByRefs({
+      entityRefs: availableOwners.map(ref =>
+        stringifyEntityRef(parseEntityRef(ref, { defaultKind: 'Group' })),
+      ),
+      fields: [
+        'kind',
+        'metadata.name',
+        'metadata.title',
+        'spec.profile.displayName',
+      ],
+    });
+    return availableOwners.map(
+      (e, i) =>
+        items.at(i) || ({ metadata: { name: e }, kind: 'Group' } as Entity),
+    );
+  }, [availableOwners]);
```

Comment:
To me this logic becomes awkward when the availableOwners are computed in one step and then used as input to a different step, and then basically thrown away.

This is made even worse by the fact that `availableOwners` contains `humanizeEntityRef` data which has destroyed a lot of the actual value of the raw data; a humanized thing should only ever be used for rendering; parsing it again should absolutely be avoided.

Seems to me that you'd want to make a _single_ `useAsync` (skip the old `useMemo`) that based on `backendEntities` computes all of the data that the rendering needs below in a single step.

Reply:
Makes sense, will update.

On a related note, the filter query param values for `owner` use the `humanizeEntityRef`'d version of the entity ref not the full reference. Is that something that is expected, especially as it should just be a rendering value? It leads to the weird logic in https://github.com/backstage/backstage/pull/16320/files#diff-4415a64d94630676f73264d7fa734e4396cdbe512042e05499f829139270d7bbR147.